#! /bin/sh /usr/share/dpatch/dpatch-run
## 03-gcc4-amd64.dpatch by Andreas Jochens <aj@andaco.de>
##
## DP: Fixes FTBS on amd64 with GCC 4

@DPATCH@

diff -Naur lush-1.1.orig/include/header.h lush-1.1/include/header.h
--- lush-1.1.orig/include/header.h	2005-01-06 02:09:38.000000000 +0000
+++ lush-1.1/include/header.h	2005-12-14 13:24:15.508117704 +0000
@@ -805,7 +805,6 @@
 #define DATE_MINUTE     4
 #define DATE_SECOND     5
 
-extern char *ansidatenames[];
 extern class date_class;
 
 TLAPI char *str_date( at *p, int *pfrom, int *pto );
--- lush-1.1.orig/src/date.c	2003-07-11 13:03:45.000000000 +0000
+++ lush-1.1/src/date.c	2005-12-14 13:24:15.509117552 +0000
@@ -311,10 +311,11 @@
   return string_buffer;
 }
 
+static at *make_date(struct date *d);
+
 static void
 date_serialize(at **pp, int code)
 {
-  static at *make_date(struct date *d);
   void *p;
 
   if (code == SRZ_READ)
diff -Naur lush-1.1.orig/src/index.c lush-1.1/src/index.c
--- lush-1.1.orig/src/index.c	2004-04-15 23:45:34.000000000 +0000
+++ lush-1.1/src/index.c	2005-12-14 13:24:15.511117248 +0000
@@ -142,6 +142,9 @@
 }
 
 
+static at *index_set(struct index*,at**,at*,int);
+static at *index_ref(struct index*,at**);
+
 static at *
 index_listeval(at *p, at *q)
 {
@@ -150,9 +153,6 @@
   register at *qsav;
   at *myp[MAXDIMS];
 
-  static at *index_set(struct index*,at**,at*,int);
-  static at *index_ref(struct index*,at**);
-
   ind = p->Object;
 
   if (ind->flags & IDF_UNSIZED)
